src/umpire/ResourceManager.hpp

```diff
@@ -132,6 +132,9 @@ class ResourceManager {
     bool hasAllocator(void* ptr);
 
     void registerAllocation(void* ptr, util::AllocationRecord* record);
+    util::AllocationRecord* findAllocationRecord(void *ptr) { 
```

Comment:
Can we make this const? Don't want people messing with records.

Reply:
CHAI is using the non-constness to update an allocation record when a user declares a raw pointer capture to be a certain length.

Reply:
I'm confused why it would be re-registered - the only reason CHAI would have to update the size is if you want to use `ResourceManager::copy` without a size argument. However, CHAI knows the size it needs to copy, so I would prefer to do it that way.